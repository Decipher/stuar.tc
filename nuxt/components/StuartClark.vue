<template>
  <div>
    <Gravatar
      class="max-w-sm rounded-lg shadow-2xl"
      email="stu@rtclark.net"
      :size="600"
    />
    <!-- @TODO - Social links -->
  </div>
</template>

<script>
import Gravatar from 'vue-gravatar'

export default {
  components: { Gravatar }
}
</script>
