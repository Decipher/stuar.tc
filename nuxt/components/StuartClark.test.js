import { mount } from '@vue/test-utils'
import StuartClark from './StuartClark.vue'

const mocks = {
  $store: {
    state: {
      config: {
        social: {
          drupal: 'Deciphered',
          github: 'Decipher',
          twitter: 'Decipher',
        },
      },
    },
  },
}
const stubs = ['LogoDrupal', 'LogoGithub', 'LogoTwitter']

describe('StuartClark', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(StuartClark, { mocks, stubs })
    expect(wrapper.vm).toBeTruthy()

    // Expect HTML to match snapshot.
    expect(wrapper.html()).toMatchSnapshot()
  })
})
