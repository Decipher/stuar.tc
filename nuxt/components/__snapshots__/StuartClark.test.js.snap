// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StuartClark is a Vue instance 1`] = `
"<div class=\\"md:pl-8\\"><img src=\\"//www.gravatar.com/avatar/d41d8cd98f00b204e9800998ecf8427e?s=600&amp;d=retro&amp;r=g\\" alt=\\"Avatar\\" class=\\"max-w-sm rounded-lg shadow-2xl w-full\\">
  <div class=\\"flex gap-4 mx-auto w-100 py-4\\"><a href=\\"https://drupal.org/u/Deciphered\\">
      <logodrupal-stub class=\\"h-10 w-10\\"></logodrupal-stub>
    </a> <a href=\\"https://github.com/Decipher\\">
      <logogithub-stub class=\\"h-10 w-10\\"></logogithub-stub>
    </a> <a href=\\"https://twitter.com/Decipher\\">
      <logotwitter-stub class=\\"h-10 w-10\\"></logotwitter-stub>
    </a></div>
</div>"
`;
